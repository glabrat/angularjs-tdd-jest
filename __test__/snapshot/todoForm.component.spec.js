import template from "../../src/components/todoForm.component.html";

describe("Render todoForm", () => {
    it("newTodo form with name data filled and submit enabled", () => {
        const componentDefinition = {
            template,
            $ctrl: {
                newTodo: {
                    name: "data"
                }
            }
        };
        expect(componentDefinition).toMatchSnapshot();
    });
    it("newTodo form with name data empty and submit disabled", () => {
        const componentDefinition = {
            template,
            $ctrl: {
                newTodo: {
                    name: ""
                }
            }
        };
        expect(componentDefinition).toMatchSnapshot();
    });
    it("newTodo form with error message displayed", () => {
        const componentDefinition = {
            template,
            $ctrl: {
                newTodo: {
                    name: ""
                },
                newtodoForm: {
                    $invalid: true,
                    todoName: {
                        $touched: true,
                        $error: { required: true }
                    }
                },
                errorMessages : {
                    name: { required: "El campo nombre es obligatorio" }
                }
            }
        };
        expect(componentDefinition).toMatchSnapshot();
    });
});
