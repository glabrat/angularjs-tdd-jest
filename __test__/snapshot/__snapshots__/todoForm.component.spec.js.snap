// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render todoForm newTodo form with error message displayed 1`] = `
<form name="$ctrl.newtodoForm" class="ng-pristine ng-scope ng-invalid ng-invalid-required">
    <p>
        <label for="todo-name">Name</label>
        <input id="todo-name" name="todoName" ng-model="[ngExpression]" required="" class="ng-pristine ng-untouched ng-empty ng-invalid ng-invalid-required">
    </p>
    <p id="todo-name-error" ng-show="[ngExpression]" class="ng-binding">
        El campo nombre es obligatorio
	</p>
    <p>
        <label for="todo-description">Description</label>
        <input id="todo-description" name="todoDescription" ng-model="[ngExpression]" class="ng-pristine ng-untouched ng-valid ng-empty">
    </p>
    <p>
        <button id="todo-submit" ng-disabled="[ngExpression]" disabled="disabled">Save</button>
    </p>
</form>
`;

exports[`Render todoForm newTodo form with name data empty and submit disabled 1`] = `
<form name="$ctrl.newtodoForm" class="ng-pristine ng-scope ng-invalid ng-invalid-required">
    <p>
        <label for="todo-name">Name</label>
        <input id="todo-name" name="todoName" ng-model="[ngExpression]" required="" class="ng-pristine ng-untouched ng-empty ng-invalid ng-invalid-required">
    </p>
    <p id="todo-name-error" ng-show="[ngExpression]" class="ng-binding ng-hide">
        
	</p>
    <p>
        <label for="todo-description">Description</label>
        <input id="todo-description" name="todoDescription" ng-model="[ngExpression]" class="ng-pristine ng-untouched ng-valid ng-empty">
    </p>
    <p>
        <button id="todo-submit" ng-disabled="[ngExpression]" disabled="disabled">Save</button>
    </p>
</form>
`;

exports[`Render todoForm newTodo form with name data filled and submit enabled 1`] = `
<form name="$ctrl.newtodoForm" class="ng-pristine ng-valid ng-scope ng-valid-required">
    <p>
        <label for="todo-name">Name</label>
        <input id="todo-name" name="todoName" ng-model="[ngExpression]" required="" class="ng-pristine ng-untouched ng-valid ng-not-empty ng-valid-required">
    </p>
    <p id="todo-name-error" ng-show="[ngExpression]" class="ng-binding ng-hide">
        
	</p>
    <p>
        <label for="todo-description">Description</label>
        <input id="todo-description" name="todoDescription" ng-model="[ngExpression]" class="ng-pristine ng-untouched ng-valid ng-empty">
    </p>
    <p>
        <button id="todo-submit" ng-disabled="[ngExpression]">Save</button>
    </p>
</form>
`;
